featurex/diagnostics/__init__.py

```diff
@@ -0,0 +1,123 @@
+import pandas as pd
+import numpy as np
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+from .collinearity import correlation_matrix
+from .collinearity import eigenvalues
+from .collinearity import condition_indices
+from .collinearity import variance_inflation_factors
+from .outliers import mahalanobis_distances
+from .validity import variances
+
+__all__ = ['collinearity', 'validity', 'outliers']
+
+class Diagnostics(object):
+    defaults = {
+        'Eigenvalues' : (lambda x: x < 0.05),
+        'ConditionIndices' : (lambda x: x > 20),
+        'VIFs' : (lambda x: x > 10),
+        'CorrelationMatrix' : (lambda x: x > 0.5),
+        'RowMahalanobisDistances' : (lambda x: x > 5),
+        'ColumnMahalanobisDistances' : (lambda x: x > 5),
+        'Variances' : (lambda x: x < 0.15)
+    }
+
+
+    ''' Class for holding diagnostics of a design matrix '''
+    def __init__(self, data, columns=None):
+        self.data = data
+
+        cols = self.data.columns if columns == None else columns
+        self.results = {}
+        self.results['Eigenvalues'] = eigenvalues(self.data[cols])
+        self.results['ConditionIndices'] = condition_indices(self.data[cols])
+        self.results['VIFs'] = variance_inflation_factors(self.data[cols])
+        self.results['CorrelationMatrix'] = correlation_matrix(self.data[cols])
+        self.results['RowMahalanobisDistances'] = mahalanobis_distances(self.data[cols])
+        self.results['ColumnMahalanobisDistances'] = mahalanobis_distances(self.data[cols], axis=1)
+        self.results['Variances'] = variances(self.data[cols])
+
+
+    def show(self, stdout=True, plot=False):
```

Comment:
Maybe rename this `summary`, for consistency with other packages (e.g., statsmodels).